cglib/src/main/java/net/sf/cglib/core/AbstractClassGenerator.java

```diff
@@ -211,7 +211,7 @@ protected Object create(Object key) {
                     cache2.put(NAME_KEY, new HashSet());
                     source.cache.put(loader, cache2);
                 } else if (useCache) {
-                    Reference ref = (Reference)cache2.get(key);
+                    Reference ref = (Reference)cache2.get(key.toString());
```

Comment:
Is `key` of `java.lang.Class` type?
If it is always a `java.lang.Class`, can you please use `#getName()` instead of `#toString()` as it would avoid producing java garbage.

Can you please add relevant comment explaining why you avoid using `key` as a, well, key?
That would make the intention clear for other developers who happen to touch this code later.